### PR TITLE
Fix SyncBatchNorm for 1.5.0

### PR DESCRIPTION
**Pull request by alsrgv**
Signed-off-by: Alex Sergeev <alexander.sergeev@live.com>

# Checklist before submitting (remove before creating the PR)

- [X] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [X] Did you update the docs?
- [X] Did you write any tests to validate this change?  
- [X] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Fixes Sync Batch Norm for PyTorch 1.5.0.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by alsrgv** [comment]
@tgaddair, this wasn't caught before because we don't test PyTorch 1.5.0 on GPU.  I think we should test 1.4.0/ 1.5.0 and PyTorch master.  What kind of test rearrangement do you have in mind for this?

File: horovod/torch/sync_batch_norm.py [hunk 1]
```diff
@@ -161,11 +165,11 @@ def backward(self, grad_output):
             sum_dy = synchronize(sum_dy_handle)
             sum_dy_xmu = synchronize(sum_dy_xmu_handle)
 
-            if _SYNC_BN_V2:
+            if _SYNC_BN_V2 or _SYNC_BN_V3:
                 mean_dy = sum_dy / count_all.sum()
                 mean_dy_xmu = sum_dy_xmu / count_all.sum()
```
Comment (thuyen) [inline horovod/torch/sync_batch_norm.py:170]: nit: we can do the `count_all.sum()` once.

**Comment by tgaddair** [comment]
> @tgaddair, this wasn't caught before because we don't test PyTorch 1.5.0 on GPU. I think we should test 1.4.0/ 1.5.0 and PyTorch master. What kind of test rearrangement do you have in mind for this?

Current GPU setup is:

```
test-gpu-openmpi-py3_6-tf1_15_0-keras2_3_1-torch1_3_0-mxnet1_4_1-pyspark2_4_0 \
test-gpu-gloo-py3_6-tf1_15_0-keras2_3_1-torch1_3_0-mxnet1_4_1-pyspark2_4_0 \
test-gpu-openmpi-gloo-py3_6-tf1_15_0-keras2_3_1-torch1_3_0-mxnet1_4_1-pyspark2_4_0 \
test-gpu-openmpi-py3_6-tf2_0_0-keras2_3_1-torch1_3_0-mxnet1_5_0-pyspark2_4_0 \
test-gpu-openmpi-py3_6-tfhead-kerashead-torchhead-mxnethead-pyspark2_4_0 \
```

I think we can bump up `tf1_15_0_*_torch1_3_0` to `tf1_15_0_*_torch_1_4_0` and `tf2_0_0_*_torch1_3_0` to `tf2_0_0_*_torch1_5_0`.  Torch master is already under test for GPU, so we should be good there.  Can you make the change?
